sites/all/modules/hr/hr_datasets/hr_datasets.module

```diff
@@ -165,5 +179,16 @@ break;
   return $block;
 }
 
+/**
+* Function to read from the HDX API and retreive results as required given a URL.
+* @param $url - this is the HDX API URL.
+* @return array
+*/
 
-
+function hr_datasets_hdx_results($url) {
```

Comment:
This is the function I was describing that we could make generic in the core feature or something, I'm sure we're doing the http_request->json_decode operation elsewhere.